Src/Support/Google.Apis.Auth/OAuth2/ImpersonatedCredential.cs

```diff
@@ -35,65 +36,102 @@ namespace Google.Apis.Auth.OAuth2
             }
         }
 
         /// <summary>
         /// Gets the source credential used to acquire the impersonated credentials.
         /// </summary>
-        public GoogleCredential SourceCredential { get; }
+        public GoogleCredential SourceCredential => HttpClientInitializers.OfType<GoogleCredential>().Single();
+
+        /// <summary>
+        /// Gets the service account to impersonate.
+        /// </summary>
+        public string TargetPrincipal { get; }
+
+        /// <summary>
+        /// Gets the chained list of delegate service accounts. May be empty.
+        /// </summary>
+        public IEnumerable<string> DelegateAccounts { get; }
+
+        /// <summary>
+        /// Gets the scopes to request during the authorization grant. May be empty.
+        /// </summary>
+        public IEnumerable<string> Scopes { get; }
 
         /// <summary>
-        /// Gets the impersonation options.
+        /// Gets for how long the delegated credential should be valid.
         /// </summary>
-        public ImpersonationOptions Options { get; }
+        public TimeSpan Lifetime { get; }
 
         /// <inheritdoc/>
-        bool IGoogleCredential.HasExplicitScopes => Options?.Scopes?.Any() == true;
+        bool IGoogleCredential.HasExplicitScopes => Scopes?.Any() == true;
 
         /// <inheritdoc/>
         bool IGoogleCredential.SupportsExplicitScopes => true;
 
-        /// <summary>Constructs a new impersonated credential using the given initializer.</summary>
-        internal ImpersonatedCredential(Initializer initializer) : base(initializer)
+        internal static ImpersonatedCredential Create(GoogleCredential sourceCredential, Initializer initializer)
         {
-            SourceCredential = initializer.SourceCredential.ThrowIfNull("initializer.SourceCredential");
-            if (!(SourceCredential.UnderlyingCredential is UserCredential || SourceCredential.UnderlyingCredential is ServiceAccountCredential))
+            initializer.ThrowIfNull(nameof(initializer));
+            sourceCredential.ThrowIfNull(nameof(sourceCredential));
+            if (!(sourceCredential.UnderlyingCredential is ServiceAccountCredential || sourceCredential.UnderlyingCredential is UserCredential))
             {
-                throw new InvalidOperationException("The underlying credential of source credential must be UserCredential or ServiceAccountCredential.");
+                throw new InvalidOperationException($"Only {nameof(ServiceAccountCredential)} and {nameof(UserCredential)} support impersonation.");
             }
-            SourceCredential = SourceCredential.CreateScoped(new string[] {GoogleAuthConsts.IamScope});
-            HttpClient.MessageHandler.Credential = SourceCredential.UnderlyingCredential as Http.IHttpExecuteInterceptor;
-            Options = initializer.Options;
+
+            initializer.HttpClientInitializers.Add(sourceCredential.CreateScoped(new string[] { GoogleAuthConsts.IamScope }));
```

Comment:
I really don't understand the initializer change (from the first commit) - let's talk about that over VC. I'm sure it's fine, but I could do with understanding it more.